### ISSUE TITLE
Looking sideways while standing in a portal does weird visual stuff

### ISSUE THREAD

**Issue by KeithKeiser971** [Gistix/portal2vr#46]
If you stand in a set of portals where they aren't both exactly vertical, and then rotate your head so that one eye is on each side of the portal, the rotation for the eyes won't match as they get snapped so that the world is always down. It would be better to base this camera snapping on the center of your headset for both eyes, or only adjust rotation once both eyes have passed through, so that they have matching rotations and don't do the split visual thing. It's hard to explain but just stand in a portal in-game and slowly pass through it sideways, you'll see what I mean. Another issue is that you can see the fake chell player model when doing this, the replica that is created on the other side of a portal as you pass through it.

**Comment by wasikuss** [comment]
example

https://github.com/Gistix/portal2vr/assets/9192409/9fa8efc3-5706-4fac-b8f0-8f0521675f5a

**Comment by HexHyte** [comment]
Duplicate of https://github.com/Gistix/portal2vr/issues/36
It's kinda funny because you said this was working for you 😕

**Comment by wasikuss** [comment]
@HexHyte wait, #36 was about not having the possibility to lean. I experienced that on 3dof. this one is completely different.

**Comment by HexHyte** [comment]
> @HexHyte wait, #36 was about not having the possibility to lean. I experienced that on 3dof. this one is completely different.

Yo the title literally says "Can't leaning **through portals**"
This is exactly what happen if you try to lean through portals, you get blocked by the wall and weird visual things happen.

**Comment by SleepingPhoenix1** [comment]
its not really a bug, it how its supposed to work. When you change dimensions with portals you see different image because head rotation doesn't change

**Comment by kwas-oc** [comment]
the real fix would be to prevent rotation of one eye when leaning through the portal. so from the point of view of 2nd portal, it would be like my head is upside down. but via this way view from eye that is on 1st portal side will align with 2nd.

even there is example where it works correctly (P - portal, | - wall) P|P
this way it works like a hole in the wall so you should expect no issues.

**Comment by wasikuss** [comment]
the real fix would be to prevent the rotation of one eye when leaning through the portal. so from the point of view of 2nd portal, it would be like my head is upside down. but via this way view from the eye that is on 1st portal side will align with 2nd.

even there is an example where it works correctly (P - portal, | - wall) P|P
this way it works like a hole in the wall so you should expect no issues.
